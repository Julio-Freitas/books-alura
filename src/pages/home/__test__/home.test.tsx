import { render, screen } from "@testing-library/react";
import { describe, it } from "vitest";
import { Home } from "..";
describe("<Home>", () => {
  it("Shoul be Home Correctly", () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByText(/Já sabe por onde começar/)).toBeInTheDocument()
  });
});
