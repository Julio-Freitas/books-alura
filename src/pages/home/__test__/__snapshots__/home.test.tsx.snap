// Vitest Snapshot v1

exports[`<Home> > Shoul be Home Correctly 1`] = `
<div>
  <div
    class="sc-lllmON loaiWC"
  >
    <div
      class="sc-bYMpWt iJymaT"
    />
    <div
      class="sc-iJnaPW cCPWzS"
    >
      <h2
        class="sc-gikAfH ifdjxU"
      >
        Já sabe por onde começar?
      </h2>
      <p
        class="sc-ezOQGI bhOvFl"
      >
        Encontre em nossa estante o que precisa para seu desenvolvimento!
      </p>
      <button
        class="sc-bcXHqe iTyHcZ"
        type="button"
      >
        Qual será sua próxima leitura?
      </button>
    </div>
    <div
      class="sc-bYMpWt dpdNOl"
    />
  </div>
  <section
    class="sc-kMjNwy kCfbbd"
  >
    <h2>
      ÚLTIMOS LANÇAMENTOS
    </h2>
    <section
      class="sc-gswNZR PXctC"
    >
      <div
        class="sc-ilhmMj hMIez"
      >
        <div />
        <div />
        <div />
      </div>
      <div
        class="sc-jIRcFI LofGU"
      >
        <header>
          <h3>
            Sobre o livro:
          </h3>
          <div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.6 377.6c-22.882 0-41.6 18.718-41.6 41.601 0 22.882 18.718 41.6 41.6 41.6s41.601-18.718 41.601-41.6c-.001-22.884-18.72-41.601-41.601-41.601zM48 51.2v41.6h41.6l74.883 151.682-31.308 50.954c-3.118 5.2-5.2 12.482-5.2 19.765 0 27.85 19.025 41.6 44.825 41.6H416v-40H177.893c-3.118 0-5.2-2.082-5.2-5.2 0-1.036 2.207-5.2 2.207-5.2l20.782-32.8h154.954c15.601 0 29.128-8.317 36.4-21.836l74.882-128.8c1.237-2.461 2.082-6.246 2.082-10.399 0-11.446-9.364-19.765-20.8-19.765H135.364L115.6 51.2H48zm326.399 326.4c-22.882 0-41.6 18.718-41.6 41.601 0 22.882 18.718 41.6 41.6 41.6S416 442.082 416 419.2c0-22.883-18.719-41.6-41.601-41.6z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 448l-30.164-27.211C118.718 322.442 48 258.61 48 179.095 48 114.221 97.918 64 162.4 64c36.399 0 70.717 16.742 93.6 43.947C278.882 80.742 313.199 64 349.6 64 414.082 64 464 114.221 464 179.095c0 79.516-70.719 143.348-177.836 241.694L256 448z"
              />
            </svg>
          </div>
        </header>
        <div
          class="sc-hhOBVt hzDgPm"
        >
          <h4>
            JavaScript Assertivo
          </h4>
          <p>
            Testes e qualidade de código em todas as camadas da aplicação
          </p>
          <span>
            Por: Gabriel Ramos
          </span>
        </div>
        <footer>
          <div>
            <span>
               A partir de:
            </span>
            <label>
               R$ 29,90 
            </label>
          </div>
          <button
            class="sc-bcXHqe GeqBS"
            type="button"
          >
            comprar
          </button>
        </footer>
      </div>
    </section>
  </section>
  <section
    class="sc-kMjNwy kCfbbd"
  >
    <h2>
      mais Vendidos
    </h2>
    <section
      class="sc-gswNZR PXctC"
    >
      <div
        class="sc-ilhmMj hMIez"
      >
        <div />
        <div />
        <div />
      </div>
      <div
        class="sc-jIRcFI LofGU"
      >
        <header>
          <h3>
            Sobre o livro:
          </h3>
          <div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.6 377.6c-22.882 0-41.6 18.718-41.6 41.601 0 22.882 18.718 41.6 41.6 41.6s41.601-18.718 41.601-41.6c-.001-22.884-18.72-41.601-41.601-41.601zM48 51.2v41.6h41.6l74.883 151.682-31.308 50.954c-3.118 5.2-5.2 12.482-5.2 19.765 0 27.85 19.025 41.6 44.825 41.6H416v-40H177.893c-3.118 0-5.2-2.082-5.2-5.2 0-1.036 2.207-5.2 2.207-5.2l20.782-32.8h154.954c15.601 0 29.128-8.317 36.4-21.836l74.882-128.8c1.237-2.461 2.082-6.246 2.082-10.399 0-11.446-9.364-19.765-20.8-19.765H135.364L115.6 51.2H48zm326.399 326.4c-22.882 0-41.6 18.718-41.6 41.601 0 22.882 18.718 41.6 41.6 41.6S416 442.082 416 419.2c0-22.883-18.719-41.6-41.601-41.6z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 448l-30.164-27.211C118.718 322.442 48 258.61 48 179.095 48 114.221 97.918 64 162.4 64c36.399 0 70.717 16.742 93.6 43.947C278.882 80.742 313.199 64 349.6 64 414.082 64 464 114.221 464 179.095c0 79.516-70.719 143.348-177.836 241.694L256 448z"
              />
            </svg>
          </div>
        </header>
        <div
          class="sc-hhOBVt hzDgPm"
        >
          <h4>
            JavaScript Assertivo
          </h4>
          <p>
            Testes e qualidade de código em todas as camadas da aplicação
          </p>
          <span>
            Por: Gabriel Ramos
          </span>
        </div>
        <footer>
          <div>
            <span>
               A partir de:
            </span>
            <label>
               R$ 29,90 
            </label>
          </div>
          <button
            class="sc-bcXHqe GeqBS"
            type="button"
          >
            comprar
          </button>
        </footer>
      </div>
    </section>
  </section>
  <div
    class="sc-gYbzsP kBMbiI"
  >
    <h4>
      CATEGORIAS MAIS BUSCADAS
    </h4>
    <label
      class="sc-dkrFOg gkBjOU"
    >
      Android
    </label>
    <label
      class="sc-dkrFOg gkBjOU"
    >
      OO
    </label>
    <label
      class="sc-dkrFOg gkBjOU"
    >
      Marketing Digital
    </label>
    <label
      class="sc-dkrFOg gkBjOU"
    >
      Agile
    </label>
    <label
      class="sc-dkrFOg gkBjOU"
    >
      Startups
    </label>
    <label
      class="sc-dkrFOg gkBjOU"
    >
      HTML & CSS
    </label>
    <label
      class="sc-dkrFOg gkBjOU"
    >
      Java
    </label>
    <label
      class="sc-dkrFOg gkBjOU"
    >
      Python
    </label>
  </div>
</div>
`;
